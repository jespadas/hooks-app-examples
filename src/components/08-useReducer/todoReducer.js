export const todoReducer = (state = [], action) => {

    /*
        The useReducer hook is used for complex state manipulations and state transitions.
        This hook function returns an array with 2 values. The first one is the state value, 
        and the second value is the dispatch function which is further used to trigger an 
        action with the help of array destructuring.
    */

    switch (action.type) {
        case 'add':
            return [...state, action.payload];

        case 'delete':
            return state.filter(todo => todo.id !== action.payload);

        case 'toggle':
            return state.map(todo =>
                (todo.id === action.payload)
                    ? { ...todo, done: !todo.done }
                    : todo
            );

        default:
            return state;
    }

}
