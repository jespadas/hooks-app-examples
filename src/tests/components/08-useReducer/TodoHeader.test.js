import { shallow } from 'enzyme';

import { TodoHeader } from "../../../components/08-useReducer/TodoHeader";
import { demoTodos } from '../../fixtures/demoTodos';

describe('Test on <TodoHeader />', () => {

    const wrapper = shallow(<TodoHeader todos={demoTodos} />);

    test('Should be displayer correctly', () => {

        expect(wrapper).toMatchSnapshot();

    })


})
