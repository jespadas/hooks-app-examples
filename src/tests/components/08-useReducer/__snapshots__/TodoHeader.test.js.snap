// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TodoHeader /> Should be displayer correctly 1`] = `
<Fragment>
  <h1>
    TodoApp (
    2
    )
  </h1>
  <hr />
</Fragment>
`;
